graph/connected_components_with_dsu.cpp

```diff
@@ -1,56 +1,112 @@
-#include <iostream>
-#include <set>
-#include <vector>
+/**
+ * @file
+ * @brief [Disjoint union](https://en.wikipedia.org/wiki/Disjoint_union)
+ *
+ * @details
+ * The Disjoint union is the technique to find connected component in graph efficiently.
+ *
+ * ### Algorithm
+ * In Graph, if you have to find out the number of connected components, there are 2 options
+ * 1. Depth first search
+ * 2. Disjoint union
+ * 1st option is inefficient, Disjoint union is the most optimal way to find this.
+ */
+#include <iostream> /// for io operations
+#include <set>    /// for std::set
+#include <vector> /// for std::vector
 
-int number_of_nodes;  // denotes number of nodes;
-std::vector<int> parent;
-std::vector<int> connected_set_size;
-void make_set() {  // function the initialize every node as it's own parent
-    for (int i = 1; i <= number_of_nodes; i++) {
+/**
+ * @namespace graph
+ * @brief Graph Algorithms
```

Comment:
```suggestion
 * @brief Graphical Algorithms
```

Reply:
> 

@ayaankhan98, this change was suggested. That is why I made. Anyway I will change it.